pkg/db-repo/change_history_manager.go

```diff
@@ -299,19 +287,97 @@ func (c *ChangeHistoryManager) rowUpdatedCondition(originalTable *tableMetadata)
 	return strings.Join(conditions, " OR ")
 }
 
-func (c *ChangeHistoryManager) updateHistoryTable(historyTable *tableMetadata) (bool, string) {
+func (c *ChangeHistoryManager) updateHistoryTable(originalTable, historyTable *tableMetadata) ([]string, error) {
+	statements := make([]string, 0)
+
+	// add new columns
 	for _, column := range historyTable.columns {
 		if column.exists {
 			continue
 		}
 
-		return true, fmt.Sprintf("ALTER TABLE %s ADD %s",
+		statements = append(statements, fmt.Sprintf("ALTER TABLE %s ADD %s",
 			historyTable.tableNameQuoted,
 			column.definition,
-		)
+		))
+	}
+
+	// remove untracked columns - without this, the triggers will not work
+	dropColumns, err := c.buildDropColumns(originalTable.tableName, historyTable.tableName)
+	if err != nil {
+		return nil, fmt.Errorf("unable to fetch columns to be dropped: %w", err)
+	}
+
+	historyColumnNames := historyTable.columnNames()
+	dropColumns = funk.Filter(dropColumns, func(column string) bool {
```

Comment:
Hmm, this is basically implementing `funk.Difference` by hand (and then just taking one of the two results). Maybe you want to go with that instead?